src/logic/data-access/cookies-provider.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+export class CookiesProvider {
+  public static getCookie(key: string): string | undefined {
+    try {
+      const filteredCookie = document.cookie
+        .split(";")
+        .filter((cookie) => cookie.split("=")[0].trim() === key);
+      const cookieValue = filteredCookie[0].split("=")[1];
```

Comment:
Skoro i tak używasz tylko pierwszej wartości, to według mnie lepiej jest użyć `find`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find